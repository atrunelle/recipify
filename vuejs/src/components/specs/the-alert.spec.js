import Vuex from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import TheAlert from '../the-alert';

describe('Component: app', () => {
  let component;
  let localVue = createLocalVue();

  localVue.use(Vuex);

  const state = {
    alert: {
      text: '',
      type: 'error',
    },
  };

  const mutations = {
    hideAlert: jest.fn(),
  };

  beforeEach(() => {
    const store = new Vuex.Store({
      state,
      mutations,
    });

    component = mount(TheAlert, {
      localVue,
      store,
    });
    jest.clearAllMocks();
  });

  it('should hide alert if no value is passed', () => {
    component.vm.onAlertInput();
    expect(mutations.hideAlert).toHaveBeenCalled();
  });

  it('should not hide alert if value is passed', () => {
    component.vm.onAlertInput({});
    expect(mutations.hideAlert).not.toHaveBeenCalled();
  });

  it('should hide success alert automatically', () => {
    global.setTimeout = (callback, delay) => callback();
    state.alert = {
      type: 'success',
      text: 'alert',
    };

    expect(mutations.hideAlert).toHaveBeenCalled();
  });

  it('should not hide alert automatically', () => {
    global.setTimeout = (callback, delay) => callback();
    state.alert = {
      type: 'error',
      text: 'alert',
    };

    expect(mutations.hideAlert).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(component.html()).toMatchSnapshot();
  });
});
