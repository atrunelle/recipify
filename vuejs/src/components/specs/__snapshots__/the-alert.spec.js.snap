// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: app should match snapshot 1`] = `
<div class="alert the-alert alert--dismissible error">
    <i aria-hidden="true" class="icon material-icons alert__icon">warning</i>
    <div>
        alert
    </div>
    <a class="alert__dismissible">
        <i aria-hidden="true" class="icon icon--right material-icons">cancel</i>
    </a>
</div>
`;
